tests/test_pipeline.py

```diff
@@ -108,6 +109,76 @@ def test_predict_all_with_novelties(self):
         self.assertEqual(self.testing_mapped_triples.shape[0], all_df['in_testing'].sum())
 
 
+class TestPipelineCheckpoints(unittest.TestCase):
+    """Test the pipeline with checkpoints."""
+
+    def setUp(self) -> None:
+        """Set up a shared result as standard to compare to."""
+        self.random_seed = 123
+        self.model = 'TransE'
+        self.dataset = 'nations'
+        self.checkpoint_name = "PyKEEN_training_loop_test_checkpoint.pt"
+        self.temporary_directory = tempfile.TemporaryDirectory()
+
+    def tearDown(self) -> None:
+        """Tear down the test case."""
+        self.temporary_directory.cleanup()
+
+    def test_pipeline_resumption(self):
+        """Test whether the resumed LCWA pipeline creates the same results as the one shot pipeline."""
+        self._test_pipeline_x_resumption(training_loop_type='LCWA')
+
+    def test_pipeline_slcwa_resumption(self):
+        """Test whether the resumed sLCWA pipeline creates the same results as the one shot pipeline."""
+        self._test_pipeline_x_resumption(training_loop_type='sLCWA')
+
+    def _test_pipeline_x_resumption(self, training_loop_type: str):
+        """Test whether the resumed pipeline creates the same results as the one shot pipeline."""
+        checkpoint_directory = self.temporary_directory.name
+
+        # TODO check that there aren't any checkpoint files already existing in the right place
```

Comment:
can we add some more tests here that illustate what's going on between each step? It's pretty high level and won't be helpful to someone without them having to fully grok the code

Reply:
Wouldn't that be the tests in *test_training.py*?

Reply:
For the same reason as given in https://github.com/pykeen/pykeen/pull/123#discussion_r534073661 , these tests only exist to see whether the pipeline handles those training loop checkpoints properly. The main functionality fully resides in the training loop.

Reply:
okay we don't have to make this part overcomplicated. Maybe a code comment saying to check the other test then?